### PR TITLE
Updates for Vasprun with MD simulations

### PR DESCRIPTION
**Pull request by gpetretto**
## Summary
I propose a few minor changes in the `Vasprun` object when parsing MD simulations and especially those produced with the ML option.

* the `converged_ionic` property was False for an MD simulation. But the check did not seems reasonable for an MD
* the number of steps for a simulation that involves ML MD is not given by the length of `ionic_steps`
* in case of NPT simulation the parsing of stresses for ML MD simulations was missing

### CODE REVIEW

File: tests/files/vasprun.xml.md
Comment (janosh) [inline tests/files/vasprun.xml.md:1]: Can you compress?

**Review by janosh — CHANGES_REQUESTED** [review]
The changes sound good to me! Thanks a lot

File: tests/files/vasprun.xml.md
Comment (janosh) [reply tests/files/vasprun.xml.md:1]: Actually, there's already `tests/files/vasprun.xml.ml_md`. Is that file incompatible with this test?

Comment (janosh) [reply tests/files/vasprun.xml.md:1]: If the new file covers a superset of test cases, you can remove the old one and plug in the path to the new one instead (only 1 test needs updating).

Comment (gpetretto) [reply tests/files/vasprun.xml.md:1]: no, the new one will not cover the ML MD, that produces the md_data attribute. The data are partially overlapping, so will not cover the same set of lines. If I need to remove one, I think that removing the `vasprun.xml.md` would be better.

Comment (janosh) [reply tests/files/vasprun.xml.md:1]: no need. where you load the file

```py
vasp_run = Vasprun(f"{TEST_FILES_DIR}/vasprun.xml.md.gz")
```

could you add a brief comment what differentiates this file from `vasprun.xml.ml_md` to leave a paper trail of our discussion in the code.

**Review by janosh — APPROVED** [review]

**Pull request merged**